.github/workflows/riot-ryot.yml

```diff
@@ -0,0 +1,94 @@
+---
+# Run 'compile_and_test_for_board.py' on connected boards.
+#
+# This workflow will run on a RYOT machine and lunch all tests on all boards
+# currently connected to that machine.
+#
+# Documentation:
+#
+# * Setup a RYOT machine:
+#   https://github.com/fjmolinas/riot-ryot/blob/master/setup.md
+#
+# * Requirements (filled by a RYOT machine):
+#   * ssh access (required secrets set)
+#   * all required flashing tools installed
+#   * udev rules that map all BOARD to /dev/riot/tty-$(BOARD)
+#   * RIOT_MAKEFILES_GLOBAL_PRE that sets PORT and DEBUG_ADAPTER_ID for each
+#     BOARD http://riot-os.org/api/advanced-build-system-tricks.html#multiple-boards-udev
+#   * Static list of connected boards or make target listing them in json format
+#     so that fromJSON can be used.
+
+name: riot-ryot
+
+on:
+  # Schedule weekly runs Saturday at 00:00 on master
+  schedule:
+    - cron: '00 0 * * 6'
+  push:
+    # Run on all new release candidates
+    tags:
+      - '[0-9][0-9][0-9][0-9].[0-9][0-9]-RC[0-9]*'
+      - '[0-9][0-9][0-9][0-9].[0-9][0-9]'
+      - '[0-9][0-9][0-9][0-9].[0-9][0-9].*'
+env:
+  # RIOT_MAKEFILES_GLOBAL_PRE to set for and DEBUG_ADAPTER_ID per BOARD
+  RIOT_MAKEFILES_GLOBAL_PRE: '/builds/conf/makefiles.pre'
+  UNIFLASH_PATH: '/opt/ti/uniflash'
+
+jobs:
+  connected_boards:
+    name: Get Connected Boards
+    runs-on: self-hosted
+    outputs:
+      boards: ${{ steps.input-boards.outputs.boards }}
+    steps:
+      # Get all currently connected boards if not passed through an input
+      - id: ci-connected-boards
+        run: echo "::set-output name=boards::$(make -C /builds/boards/ list-boards-json --no-print-directory)"
+
+  # Runs all tests on connected boards
+  compile_and_test_for_board:
+    name: ${{ matrix.board }}
+    runs-on: self-hosted
+    needs: connected_boards
+    # ci-riot-tribe has 8 cores, parallelism will depend on actually configured
+    # runners
+    strategy:
+      max-parallel: 7
+      fail-fast: false
+      matrix:
+        board: ${{ fromJson(needs.connected_boards.outputs.boards) }}
+    env:
+      BUILD_IN_DOCKER: 1
+      COMPILE_AND_TEST_FOR_BOARD: /builds/boards/RIOT/dist/tools/compile_and_test_for_board/compile_and_test_for_board.py
+      # args for compile_and_test_for_board script
+      COMPILE_AND_TEST_ARGS: --with-test-only --report-xml --incremental
```

Comment:
Can github parse the xml report and show individual tests?
This could be a nice future improvement.

Reply:
Would be nice in general (`test-on-iotlab` and `release-tests` also provide xUnit XML). But I don't think that GitHub supports parsing for that (in `release-tests` I used `junit2html` to generate some HTML in the ZIP'd artifacts). There where some (inofficial) plans to maybe fold all our xUnit XML into what is now https://hil.riot-os.org, but I think that is mostly abandoned for now. Can you confirm or deny that @MrKevinWeiss?

Reply:
Currently the https://hil.riot-os.org page uses xunit based xml file to generate the overview but uses conventions (ie board_app) to do so and I don't think is meant for this.

I was thinking of throwing all the xunit things into a database type collector and create a more generic way of dealing with the data (ie ELK stack or graphana based thingy). This is where I think it would be nice to converge.

The problem is actually doing it and maintaining it...

It seems junit is integrated with Jenkins OK but I really don't know if it plays well with github actions.  There is https://github.com/marketplace/actions/publish-unit-test-results but I have not used it.

Reply:
> It seems junit is integrated with Jenkins OK but I really don't know if it plays well with github actions. There is https://github.com/marketplace/actions/publish-unit-test-results but I have not used it.

I'd say: let's try it e.g. for the release tests!

Reply:
Sadly, this only works for PRs https://github.com/miri64/RIOT/runs/2287794648?check_suite_focus=true#step:5:17